src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -50,7 +54,16 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                try
```

Comment:
This can be done in the beginning to avoid creating a ConfigurationClient that will never be used.

Reply:
I think we need to pass in a non-null client to the provider constructor, is there a way around that?

Reply:
I was thinking of something like this:

```cs
public IConfigurationProvider Build(IConfigurationBuilder builder)
{
    try
    {
        string envType = Environment.GetEnvironmentVariable(RequestTracingConstants.AspNetCoreEnvironmentVariable) ??
                            Environment.GetEnvironmentVariable(RequestTracingConstants.DotNetCoreEnvironmentVariable);
        if (!envType.Equals(_environmentName, StringComparison.OrdinalIgnoreCase))
        {
            return new EmptyConfigurationProvider();
        }
    }
    catch (SecurityException) 
    {
        // cant read environment variables - assume prod
        if (!_environmentName.Equals(Environments.Production, StringComparison.OrdinalIgnoreCase))
        {
            return new EmptyConfigurationProvider();
        }
    }

    IConfigurationProvider provider = null;

    try
    {

        ... // no change needed
    }
    catch (InvalidOperationException e)
    {
        ... // no change needed
    }

    return provider ?? new EmptyConfigurationProvider();
}
```

Reply:
Oh thanks, forgot about using empty provider